src/ILCompiler.Compiler/src/Compiler/ExportedMethodsRootProvider.cs

```diff
@@ -12,10 +15,20 @@ namespace ILCompiler
     public class ExportedMethodsRootProvider : ICompilationRootProvider
     {
         private EcmaModule _module;
+        private List<EcmaMethod> _methods;
 
         public ExportedMethodsRootProvider(EcmaModule module)
         {
             _module = module;
+            _methods = new List<EcmaMethod>();
+        }
+
+        public IEnumerable<EcmaMethod> ExportedMethods
+        {
+            get
+            {
+                return _methods;
```

Comment:
Could you factor this class in a way that we can't get different results depending on when `ExportedMethods` is accessed? It's not a problem right now, but I can see someone getting burned by this at some point.